src/test/scala/sangria/relay/ConnectionWithCustomPageInfoSpec.scala

```diff
@@ -0,0 +1,205 @@
+package sangria.relay
+
+import org.scalatest.{ Matchers, WordSpec }
+import sangria.execution.Executor
+import sangria.parser.QueryParser
+import sangria.relay.util.AwaitSupport
+import sangria.schema._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.Success
+
+class ConnectionWithCustomPageInfoSpec extends WordSpec with Matchers with AwaitSupport {
+
+  /**
+   * custom PageInfo to be included in Connection
+   */
+  case class CustomPageInfo(
+    currentPage: Int,
+    hasNextPage: Boolean = false,
+    hasPreviousPage: Boolean = false,
+    startCursor: Option[String] = None,
+    endCursor: Option[String] = None) extends PageInfo
+
+  /**
+   * custom Connection includes CustomPageInfo
+   */
+  case class CustomConnection[T, P <: PageInfo](
+    pageInfo: CustomPageInfo,
+    edges: Seq[Edge[T]]) extends Connection[T, CustomPageInfo]
```

Comment:
Instead of using inheritance (extending `Connection`), maybe you can use type-class based mechanism to identify `CustomConnection` as a `ConnectionLike`?

Reply:
Thanks! I fixed it using `ConnectionLike` type-class.